cmd/snap/cmd_run.go

```diff
@@ -1014,16 +1014,18 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 	}
 	if info.Base != "" {
 		cmd = append(cmd, "--base", info.Base)
-	}
-	// kernels have no explicit base, we use the boot base
-	if info.Type() == snap.TypeKernel && info.Base == "" {
-		modelAssertion, err := x.client.CurrentModelAssertion()
-		if err != nil {
-			return err
-		}
-		modelBase := modelAssertion.Base()
-		if modelBase != "" {
-			cmd = append(cmd, "--base", modelBase)
+	} else {
+		if info.Type() == snap.TypeKernel {
+			// kernels have no explicit base, we use the boot base
+
+			modelAssertion, err := x.client.CurrentModelAssertion()
+			if err != nil {
+				return fmt.Errorf("cannot get model assertion for kernel hook: %v", err)
```

Comment:
this code is shared so we are not sure it's a hook? though we don't expect but don't block kernel apps?

maybe "cannot get model assertion to setup kernel app|hook run: %v"

where we choose app or hook based on hook != "" ?